### PR TITLE
[Android] Fix Android commissioning complete fail

### PR DESCRIPTION
**Pull request by joonhaengHeo**
#### Problem
What is being fixed?  Examples:
* Fix Android commissioning complete fail. (invalid keypair).
* This issue only occurs when no certificate is set.
* this issue was caused by the fix in #19545.

#### Change overview
The location of the keypair variable declaration has been changed.

#### Testing
How was this tested? (at least one bullet point required)
* Check Commissioning using Android Device.

### CODE REVIEW

File: src/controller/java/AndroidDeviceControllerWrapper.cpp [hunk 1]
```diff
@@ -193,6 +193,8 @@ AndroidDeviceControllerWrapper * AndroidDeviceControllerWrapper::AllocateNew(
     }
     MutableByteSpan rcacSpan(rcac.Get(), kMaxCHIPDERCertLength);
 
+    Crypto::P256Keypair ephemeralKey;
+
```
Comment (mrjerryjohns) [inline src/controller/java/AndroidDeviceControllerWrapper.cpp:197]: Can you describe why changing the scope here fixes things? As far as I can tell, nothing in the first `if {..}` utilizes `ephemeralKey`?

Comment (bzbarsky-apple) [reply src/controller/java/AndroidDeviceControllerWrapper.cpp:197]: `setupParams` stores a pointer to the key, and is used after the else.  So this is not about the `if`; it's about the code that ran after `ephemeralKey` went out of scope but was using a pointer to it.

The real right fix here is that `setupParams` should be declared after all the things it's going to store pointers to, including `ephemeralKey`....

**Review by bzbarsky-apple — APPROVED** [review]

File: src/controller/java/AndroidDeviceControllerWrapper.cpp (hunk 1 above)
Comment (joonhaengHeo) [reply src/controller/java/AndroidDeviceControllerWrapper.cpp:198]: @andy31415 I added comment. Thank you for your comment.
@mrjerryjohns Someone else has already mentioned my point.
@bzbarsky-apple I thought about a better way, but couldn't find a better way, so I modified it like this.

**Comment by woody-apple** [comment]
Fast tracking platform changes.

**Review by andy31415 — APPROVED** [review]

**Pull request merged**